share/spice/timer/timer.js

```diff
@@ -406,19 +440,31 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
     });
 
     env.ddg_spice_timer = function(api_result) {
+        var timers = [],
+            timerInterval,
+            $addTimerBtn;
+
+        function addTimer(startingTime) {
+            var timer = new Timer(timers.length + 1, startingTime);
+            timer.$element.insertBefore($addTimerBtn.parent());
+            timers.push(timer);
+        }
         
         function onShow() {
```

Comment:
@russellholt I thought I had seen it in more Spices, but it looks like right now it's only Amazon: 

https://github.com/duckduckgo/zeroclickinfo-spice/blob/4e967c6672c9aeaadb54652502270dd1ce159c57/share/spice/amazon/amazon.js#L50

Almost every Spice that defines an `onShow` really only needs it to run once though, and right now they're running every time we switch back to the tab. In many cases we're using it to add classes to divs, define handlers for events, or load more data -- all of which should only be done once.

For example, [movies](https://duckduckgo.com/?q=movies&t=ffab&ia=movies) is loading the images for each tile from a 3rd party in `onItemShown` -- I just realized it's running each time we switch to the tab, causing a lot of unnecessary requests. 

---

For Goodies, we're using a check to make sure it only runs once for CheatSheet.js (https://github.com/duckduckgo/zeroclickinfo-goodies/blob/2173d26caf4a33480ea7518d3c18c666e6de5267/share/goodie/cheat_sheets/cheat_sheets.js#L91) and we'll need it for the 2048 game (still in-progress), to bind our event-handlers to DOM elements